src/main/io/gps.c

```diff
@@ -500,17 +656,28 @@ static void ubloxSendMessage(const uint8_t *data, uint8_t len)
     // Save state for ACK waiting
     gpsData.ackWaitingMsgId = data[3]; //save message id for ACK
     gpsData.ackTimeoutCounter = 0;
-    gpsData.ackState = UBLOX_ACK_WAITING;
+    gpsData.ackState = skipAck ? UBLOX_ACK_GOT_ACK : UBLOX_ACK_WAITING;
+}
+
+static void ubloxSendClassMessage(ubxProtocolBytes_e class_id, ubxProtocolBytes_e msg_id, uint16_t length)
+{
+    ubxMessage_t tx_buffer;
+    tx_buffer.header.preamble1 = PREAMBLE1;
+    tx_buffer.header.preamble2 = PREAMBLE2;
+    tx_buffer.header.msg_class = class_id;
+    tx_buffer.header.msg_id = msg_id;
+    tx_buffer.header.length = length;
+    ubloxSendMessage((const uint8_t *) &tx_buffer, length + 6, false);
 }
 
-static void ubloxSendConfigMessage(ubxMessage_t *message, uint8_t msg_id, uint8_t length)
+static void ubloxSendConfigMessage(ubxMessage_t *message, uint8_t msg_id, uint8_t length, bool skipAck)
```

Comment:
Agreed this could be cleaned up, there are too many layers of helper functions, some of which only add a value for one of the fields (like the example above that just sets `message->header.msg_class = CLASS_CFG;` and passes everything else through.

Reply:
It would be good to have just one `ubloxSendClassMessage()` function to cover both kinds of message.
I can't code that, it's too complex for me.
Could this be fixed in a separate PR?

Reply:
It just about removing duplicate code, moving it to cleanup PR is good idea.